funcx_sdk/funcx/sdk/executor.py

```diff
@@ -0,0 +1,249 @@
+import time
+import threading
+import os
+import uuid
+import sys
+from concurrent.futures import Future
+import concurrent
+import logging
+import asyncio
+import websockets
+import json
+import dill
+from websockets.exceptions import InvalidHandshake
+import multiprocessing as mp
+import atexit
+
+from funcx.sdk.asynchronous.ws_polling_task import WebSocketPollingTask
+logger = logging.getLogger("asyncio")
+
+
+class AtomicController():
+
+    def __init__(self, start_callback, stop_callback, init_value=0):
+        self._value = 0
+        self.lock = threading.Lock()
+        self.start_callback = start_callback
+        self.stop_callback = stop_callback
+
+    def increment(self):
+        with self.lock:
+            if self._value == 0:
+                self.start_callback()
+            self._value += 1
+
+    def decrement(self):
+        with self.lock:
+            self._value -= 1
+            if self._value == 0:
+                self.stop_callback()
+            return self._value
+
+    def value(self):
+        with self.lock:
+            return self._value
+
+    def __repr__(self):
+        return f"AtomicController value:{self._value}"
+
+
+class FuncXExecutor(concurrent.futures.Executor):
+    """ An executor
+    """
+
+    def __init__(self, funcx_client,
+                 results_ws_uri: str = 'ws://localhost:6000',
```

Comment:
Is there a default ws address for production?

Reply:
I've updated this.